File: Actito.Scannables.iOS.Binding.Bridge/Sources/Models/ActitoScannable.swift
import Foundation
import ActitoScannablesKit
import ActitoBinding

@objc
public class ActitoScannable : NSObject {
    @objc public let scannableId: String
    @objc public let name: String
    @objc public let tag: String
    @objc public let type: String
    @objc public let notification: ActitoNotification?

    @objc public init(scannableId: String, name: String, tag: String, type: String, notification: ActitoNotification?) {
        self.scannableId = scannableId
        self.name = name
        self.tag = tag
        self.type = type
        self.notification = notification
    }

    public convenience init(from scannable: ActitoScannablesKit.ActitoScannable) {
        self.init(
            scannableId: scannable.id,
            name: scannable.name,
            tag: scannable.tag,
            type: scannable.type,
            notification: scannable.notification.map { ActitoNotification(from: $0) }
        )
    }
}

File: Actito.Scannables.Core/Models/ActitoScannable.cs
using ActitoSdk.Core.Models;

namespace ActitoSdk.Scannables.Core.Models;

public class ActitoScannable
{
    public string Id { get; }
    public string Name { get; }
    public string Tag { get; }
    public string Type { get; }
    public ActitoNotification? Notification { get; }

    public ActitoScannable(string id, string name, string tag, string type, ActitoNotification? notification)
    {
        Id = id;
        Name = name;
        Tag = tag;
        Type = type;
        Notification = notification;
    }
}
